agent/xds/listeners_apigateway.go

```diff
@@ -35,13 +32,19 @@ func (s *ResourceGenerator) makeAPIGatewayListeners(address string, cfgSnap *pro
 		listenerKey := readyListener.listenerKey
 		boundListener := readyListener.boundListenerCfg
 
-		var certs []structs.InlineCertificateConfigEntry
+		// Collect the referenced certificate config entries
+		var certs []structs.ConfigEntry
 		for _, certRef := range boundListener.Certificates {
-			cert, ok := cfgSnap.APIGateway.Certificates.Get(certRef)
-			if !ok {
-				continue
+			switch certRef.Kind {
+			case structs.InlineCertificate:
+				if cert, ok := cfgSnap.APIGateway.Certificates.Get(certRef); ok {
+					certs = append(certs, cert)
+				}
```

Comment:
Did you intentionally omit this check in both cases?
```
if !ok {
    continue
}
```

Reply:
Yes because I think this would only be a programmer error, not something that people should run into.

Reply:
they're functionally the same unless I'm missing something.

```go
if ok {
 // do something
}
```
vs.

```go
if !ok {
  continue
}

// do something
```

Reply:
I guess it just depends on the behaviour we want
```
if ok { 
    // do something 
}
```
using the above code, we successfully retrieve the certificate and move on to the next thing. 
```
if !ok {
  continue
}
// do something
```
This code here indicates that if we cert is found, we can skip the rest of the iteration and move onto the next thing. Just depends on how we want to handle it. I saw that we used `if !ok { continue}` in the previous case and was wonder why we didn't here.

Reply:
ohhhh I see what you're saying. I just went with the current flavor because it simplified the code and had the same outcome, no more reasoning that